src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractJenkinsManifestMojo.java

```diff
@@ -72,6 +72,15 @@
     @Parameter(required = true)
     protected String minimumJavaVersion;
 
+    /**
+     * Indicates the plugin should not get any implied dependency until the specified (Jenkins core) version included.
+     * <p>Example: if a plugin specifies this with a value of <em>2.60.3</em>, and used bouncycastle API (which got detacted in 2.16) without explictly
+     * declaring the dependency against bouncycastle-api, then the plugin would likely fail at runtime because Jenkins Core Plugin Management will
+     * then <strong>not</strong> add bouncycastle-api as an implicit dependency.</p>
+     */
+    @Parameter(property = "hpi.noDetachedPluginUntil", required = true)
```

Comment:
Why `required`? This certainly sounds like it should be optional to me.

Reply:
The intent is to have it advertised once we propagate it at some point in the plugin-pom, so plugin developers are hinted about that when bumping the parent.

We'd document there something like:

> If you want all implied dependencies, and want to go yolo: just put `1.0` and move on.
> Better: do X and Y do to actually be able to define a sensible value.

And worst worst case: we put 1.0 as default value in `plugin-pom`.

Reply:
Uh, having this new manifest attribute be `1.0`, or generally _less than_ `Jenkins-Version`, seems rather confusing to me. I would really recommend the attribute be omitted unless you are actually using this feature. Would suffice to make this parameter optional, and treat null or `""` as unset. I am not even sure this is something we want to advertise in `plugin-pom`; the small number of plugins which need this could define `hpi.noDetachedPluginUntil`, and `buildPlugin.groovy` could detect this and automatically include that version in the testing matrix.